### PR TITLE
Add blocked state as SyncJobStatus.

### LINKED ISSUE #2538: Do you think it would be important to handle [WorkInfo.State#BLOCKED](https://developer.android.com/reference/androidx/work/WorkInfo.State#BLOCKED) status or defaulting to error would just be enough?
Do you think it would be important to handle [WorkInfo.State#BLOCKED](https://developer.android.com/reference/androidx/work/WorkInfo.State#BLOCKED) status or defaulting to error would just be enough?

_Originally posted by @ellykits in https://github.com/google/android-fhir/pull/2511#discussion_r1588948160_

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #2538

**Description**
Added Blobked state to SyncJobStatus

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: engine/src/androidTest/java/com/google/android/fhir/sync/SyncInstrumentedTest.kt [hunk 1]
```diff
@@ -104,12 +105,49 @@ class SyncInstrumentedTest {
     assertThat(states.last()).isInstanceOf(CurrentSyncJobStatus.Succeeded::class.java)
   }
 
+  @Test
+  fun oneTime_worker_nextExecutionAfterSucceeded() {
```
Comment (MJ1998) [inline engine/src/androidTest/java/com/google/android/fhir/sync/SyncInstrumentedTest.kt:109]: What is this function trying to test ?
Maybe name the function in this format <context>_<action>_<result>. For example `newPatient_insert_recordIsAddedToDatabase`

Comment (santosh-pingle) [reply engine/src/androidTest/java/com/google/android/fhir/sync/SyncInstrumentedTest.kt:109]: Lets review #2511 first, its parent pr.

**Comment by santosh-pingle** [comment]
precondition : review and  merge #2511

**Comment by santosh-pingle** [comment]
After merging master it shows individual commit in the log, as result it shows as many as files as changed files.
Hence closed this pr and created new one #2607

**Pull request closed without merging**